admin/class-create-block-theme-admin.php

```diff
@@ -331,6 +336,18 @@ function add_theme_json_to_local ( $export_type ) {
 		);
 	}
 
+	function add_theme_json_variation_to_local ( $export_type, $theme ) {
+		$variation_path = get_stylesheet_directory() . DIRECTORY_SEPARATOR . 'styles' . DIRECTORY_SEPARATOR;
+		if ( ! file_exists( $variation_path ) ) {
+			mkdir( $variation_path, 0755, true );
+		}
+
+		file_put_contents(
+			$variation_path . '/' . $theme['variation'] . '.json',
+			MY_Theme_JSON_Resolver::export_theme_data( $export_type )
```

Comment:
Should we add the `title` property to the new json file? It doesn't look like we have to, as the name is pulled from the filename in global styles:

![image](https://user-images.githubusercontent.com/1645628/181209075-c3968b02-7c32-4470-9ee7-b61c702c00fa.png)

But then I'm not sure what the `title` property is for! Maybe so you can format the title differently from the filename? So in this case we could set this to the original variation name that's entered.

Reply:
I think changing the content of the generated json will bring extra complexity, but sanitizing the slug makes a lot of sense. We can revisit the title of it in a follow up maybe?